src/BenchmarksApps/SslStream/Client/ClientOptions.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.CommandLine;
+using System.CommandLine.Binding;
+using System.Net.Security;
+using System.Security.Authentication;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using SslStreamCommon;
+
+namespace SslStreamClient;
+
+public enum CertificateSource
+{
+    Context,
+    Certificate,
+    Callback,
+}
+
+public class ClientOptions : OptionsBase
+{
+    public string Hostname { get; set; } = null!;
+    public CertificateSource CertificateSource { get; set; }
+    public X509Certificate2? ClientCertificate { get; set; } = null!;
+    public string? TlsHostName { get; set; }
+    public Scenario Scenario { get; set; }
+    public TimeSpan Warmup { get; set; }
+    public TimeSpan Duration { get; set; }
+}
+
+public class OptionsBinder : BinderBase<ClientOptions>
```

Comment:
I think I had to use the explicit binder in HttpClient benchmarks, because I had to manually parse the headers parameter from `string[]` into the list of (name, value) pairs... I think it should be possible to use an automatic binder in your case.

Reply:
as part of the binding, the code loads up the X509Certificate (path + optional password), so that the instance is already part of the resulting ClientOptions.